### ISSUE TITLE
Crash in Player Tick Event kicks players in specific locations

### ISSUE THREAD

**Issue by CovertJaguar** [Glitchfiend/ToughAsNails#255]
Version 1.10.2 - 1.1.1

```
[23:16:17] [Server thread/ERROR] [FML/]: Exception caught during firing event net.minecraftforge.fml.common.gameevent.TickEvent$PlayerTickEvent@31bc268a:
java.lang.NullPointerException
	at toughasnails.temperature.modifier.ObjectProximityModifier.getBlockTemperature(ObjectProximityModifier.java:95) ~[ObjectProximityModifier.class:?]
	at toughasnails.temperature.modifier.ObjectProximityModifier.modifyChangeRate(ObjectProximityModifier.java:47) ~[ObjectProximityModifier.class:?]
	at toughasnails.temperature.TemperatureHandler.update(TemperatureHandler.java:93) ~[TemperatureHandler.class:?]
	at toughasnails.handler.ExtendedStatHandler.onPlayerTick(ExtendedStatHandler.java:69) ~[ExtendedStatHandler.class:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_341_ExtendedStatHandler_onPlayerTick_PlayerTickEvent.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) ~[ASMEventHandler.class:?]
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185) [EventBus.class:?]
	at net.minecraftforge.fml.common.FMLCommonHandler.onPlayerPostTick(FMLCommonHandler.java:365) [FMLCommonHandler.class:?]
	at uk.co.mysterymayhem.gravitymod.asm.EntityPlayerWithGravity.func_184808_cD(EntityPlayerWithGravity.java:722) [EntityPlayerWithGravity.class:?]
	at net.minecraft.entity.player.EntityPlayer.func_70071_h_(EntityPlayer.java:292) [zs.class:?]
	at net.minecraft.entity.player.EntityPlayerMP.func_71127_g(EntityPlayerMP.java:303) [lu.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_73660_a(NetHandlerPlayServer.java:162) [me.class:?]
	at net.minecraftforge.fml.common.network.handshake.NetworkDispatcher$1.func_73660_a(NetworkDispatcher.java:213) [NetworkDispatcher$1.class:?]
	at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:287) [eo.class:?]
	at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:180) [md.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:732) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:387) [ld.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:613) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:471) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
[23:16:17] [Server thread/ERROR] [FML/]: Index: 4 Listeners:
[23:16:17] [Server thread/ERROR] [FML/]: 0: HIGHEST
[23:16:17] [Server thread/ERROR] [FML/]: 1: ASM: cz.hanakocz.rccosmetic.events.PlayerEventListener@3c32c8a5 onPlayerTick(Lnet/minecraftforge/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[23:16:17] [Server thread/ERROR] [FML/]: 2: ASM: uk.co.mysterymayhem.gravitymod.common.listeners.GravityManagerCommon@3b93580d onPlayerUpdateTickStart(Lnet/minecraftforge/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[23:16:17] [Server thread/ERROR] [FML/]: 3: NORMAL
[23:16:17] [Server thread/ERROR] [FML/]: 4: ASM: toughasnails.handler.ExtendedStatHandler@8087982 onPlayerTick(Lnet/minecraftforge/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[23:16:17] [Server thread/ERROR] [FML/]: 5: ASM: toughasnails.handler.health.MaxHealthHandler@37ca1a11 onPlayerLogin(Lnet/minecraftforge/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[23:16:17] [Server thread/ERROR] [FML/]: 6: ASM: pl.asie.lib.tweak.enchantment.EnchantmentTweak@5e450351 enchEvent(Lnet/minecraftforge/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[23:16:17] [Server thread/ERROR] [FML/]: 7: ASM: xreliquary.handler.CommonEventHandler@42a96004 onPlayerTick(Lnet/minecraftforge/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[23:16:17] [Server thread/ERROR] [FML/]: 8: ASM: uk.co.mysterymayhem.gravitymod.common.listeners.FallOutOfWorldUpwardsListenerCommon@48b589f5 onPlayerTick(Lnet/minecraftforge/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[23:16:17] [Server thread/ERROR] [FML/]: 9: ASM: ic2.core.item.armor.jetpack.JetpackHandler@7b4df8a8 tick(Lnet/minecraftforge/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[23:16:17] [Server thread/ERROR] [FML/]: 10: LOWEST
[23:16:17] [Server thread/ERROR] [FML/]: 11: ASM: uk.co.mysterymayhem.gravitymod.common.listeners.GravityManagerCommon@3b93580d onPlayerUpdateTickEnd(Lnet/minecraftforge/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
```

**Comment by Adubbz** [comment]
That class has undergone quite a few changes recently, so I'd need confirmation on what you're currently using (otherwise the line numbers don't make much sense). Also, is it definitely not because of the modifications made? (I only ask because the past few issues were :P )

**Comment by CovertJaguar** [comment]
As far as I know we haven't touched that. Its 1.1.1.

**Comment by hrudgar** [comment]
Getting the same issue on 1.10.2, and I don't have the gravity mod installed CJ does:

```
java.lang.NullPointerException: Ticking player
    at toughasnails.temperature.modifier.ObjectProximityModifier.getBlockTemperature(ObjectProximityModifier.java:95)
    at toughasnails.temperature.modifier.ObjectProximityModifier.modifyChangeRate(ObjectProximityModifier.java:47)
    at toughasnails.temperature.TemperatureHandler.update(TemperatureHandler.java:93)
    at toughasnails.handler.ExtendedStatHandler.onPlayerTick(ExtendedStatHandler.java:69)
    at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_322_ExtendedStatHandler_onPlayerTick_PlayerTickEvent.invoke(.dynamic)
    at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
    at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185)
    at net.minecraftforge.fml.common.FMLCommonHandler.onPlayerPostTick(FMLCommonHandler.java:365)
    at net.minecraft.entity.player.EntityPlayer.func_184808_cD(EntityPlayer.java:382)
    at net.minecraft.entity.player.EntityPlayer.func_70071_h_(EntityPlayer.java:292)
    at net.minecraft.entity.player.EntityPlayerMP.func_71127_g(EntityPlayerMP.java:303)
    at net.minecraft.network.NetHandlerPlayServer.func_73660_a(NetHandlerPlayServer.java:162)
    at net.minecraftforge.fml.common.network.handshake.NetworkDispatcher$1.func_73660_a(NetworkDispatcher.java:213)
    at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:287)
    at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:180)
    at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:732)
    at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:613)
    at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:240)
    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:471)
    at java.lang.Thread.run(Thread.java:745)
```

[Pastebin to the full crashlog](https://pastebin.com/WRD4d1G3)

Seems to be pretty random, but the two times it has happened, it's been the dead of winter (Dead Crops on the ground). In an artic biome in Creative mode.

**Comment by Adubbz** [comment]
This has been fixed in 1.11 and 1.12, if someone wants to submit a PR for 1.10 they're certainly welcome to

**Comment by Adubbz** [comment]
Merged the PR for 1.10 as well, so this should be fixed across the board now
